### PR TITLE
Provide site failover functionality

### PR DESCRIPTION
**Pull request by krishnasrinivas**
This is a breaking change in terms of handling of CLI arguments.

### CODE REVIEW

File: ui.go [hunk 1]
```diff
@@ -63,9 +63,14 @@ func initNodesTable() *nodesView {
 		"MAX LATENCY"}
 
 	return &nodesView{t, header}
-
 }
-func initUI(backends []*Backend) {
+
+func initUI(lb *lb) {
```
Comment (harshavardhana) [inline ui.go:68]: in the UI can we separately show two sites? @krishnasrinivas

File: main.go [hunk 2]
```diff
@@ -605,12 +640,16 @@ func main() {
...
-  sidekick [FLAGS] ENDPOINTs...
-  sidekick [FLAGS] ENDPOINT{1...N}
+  sidekick [FLAGS] SITE1 [SITE2..]
 
 FLAGS:
   {{range .VisibleFlags}}{{.}}
   {{end}}
+
+SITE:
+Each SITE is a comma seperated list of zones of that site: http://172.17.0.{2..5},http://172.17.0.{6..9}
```
Comment (poornas) [inline main.go:650]:
```suggestion
Each SITE is a comma separated list of zones of that site: http://172.17.0.{2..5},http://172.17.0.{6..9}
```

**Comment by krishnasrinivas** [comment]
@poornas there is a gap between the two sites, I could not figure out the bug, can you take a look?
```
sidekick --health-path "/minio/health/ready" http://172.17.0.{2...5}:9000 http://172.17.0.{6...9}:9000
```
2020-05-27-145459_1318x365_scrot

File: ui.go (hunk 1 above)
Comment (krishnasrinivas) [reply ui.go]: done, there is a formatting issue in the grid output, checking with @poornas

**Comment by poornas** [comment]
> @poornas there is a gap between the two sites, I could not figure out the bug, can you take a look?
> 
> ```
> sidekick --health-path "/minio/health/ready" http://172.17.0.{2...5}:9000 http://172.17.0.{6...9}:9000
> ```
> 
> 2020-05-27-145459_1318x365_scrot

The grid's rowspan needs to be adjusted - currently it is fixed at 20 since it was just a single table. If multiple sites need to fit into the same ui, you will need to calculate how many entries to display per site and adjust the rowspan accordingly.

File: README.md [hunk 3]
```diff
@@ -40,6 +39,10 @@ FLAGS:
   --debug                            output verbose trace
   --help, -h                         show help
   --version, -v                      print the version
+
+SITE:
+Each SITE is a comma seperated list of zones of that site: http://172.17.0.{2..5},http://172.17.0.{6..9}
```
Comment (poornas) [inline README.md:44]:
```suggestion
Each SITE is a comma separated list of zones of that site: http://172.17.0.{2..5},http://172.17.0.{6..9}
```

**Comment by harshavardhana** [comment]
BTW can we adapt the spark operator example with breaking changes as well?

**Comment by krishnasrinivas** [comment]
Having two tables, one for each site was causing alignment and flickering issues.
Hence instead I have made it one table like before, but additional column of site number.

> BTW can we adapt the spark operator example with breaking changes as well?

Spark example had one zone, so no change needed. Did you mean add multi-site failover example for spart? I think it's OK not to and keep the spark example simple. (Since for spark site failover will not be useful - i.e going over WAN link)

File: README.md (hunk 3 above)
Comment (poornas) [reply README.md:44]: can you fix the typo?

**Comment by poornas** [comment]
Screenshot from 2020-05-28 13-58-28

The sidekick logo is hidden now - if terminal is not wide enough. Needs to be fixed.

**Comment by harshavardhana** [comment]
> Spark example had one zone, so no change needed. Did you mean add multi-site failover example for spart? I think it's OK not to and keep the spark example simple. (Since for spark site failover will not be useful - i.e going over WAN link)

Okay :+1: @krishnasrinivas

**Comment by krishnasrinivas** [comment]
> The sidekick logo is hidden/overlapping with rest of the table- if terminal is not wide enough. Needs to be fixed.

@poornas in this code only sidekick logo gets hidden, in the master branch I see that entire table vanishes. So this is an improvement over the master branch. We can fix the sidekick logo vanishing later as a lower priority as I am unable to figure out how to fix it.

**Pull request merged**